src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -93,6 +97,42 @@ def _dist_count_aggregation_on_tx_status_factory(
     )
 
 
+def _aggregation_on_tx_satisfaction_func_factory(aggregate):
```

Comment:
It probably makes sense to have one factory capable of handling the functionality of this and `_aggregation_on_tx_status_func_factory` factory above. I'll follow up with a PR for a generalized factory in the future.